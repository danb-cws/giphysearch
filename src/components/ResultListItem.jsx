import React from "react";

const ResultListItem = ({ gif, selectedItem, index, onItemSelect }) => {
  return (
    <li>
      <a
        className={selectedItem === index ? "selected-item" : null}
        onClick={() => {
          onItemSelect(index);
        }}
      >
        <h5>{gif.title}</h5>
        <img
          src={gif.images.fixed_height_small_still.url}
          width={gif.images.fixed_height_small_still.width}
          height={gif.images.fixed_height_small_still.height}
          alt={gif.title}
        />
      </a>
    </li>
  );
};

export default ResultListItem;
