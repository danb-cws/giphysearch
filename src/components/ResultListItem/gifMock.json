{
  "type": "gif",
  "id": "3ohc1f1vrhZsufbBcI",
  "slug": "fun-party-90s-3ohc1f1vrhZsufbBcI",
  "url": "https://giphy.com/gifs/fun-party-90s-3ohc1f1vrhZsufbBcI",
  "bitly_gif_url": "https://gph.is/2yYUlzy",
  "bitly_url": "https://gph.is/2yYUlzy",
  "embed_url": "https://giphy.com/embed/3ohc1f1vrhZsufbBcI",
  "username": "ladypat",
  "source": "",
  "rating": "pg",
  "content_url": "",
  "source_tld": "",
  "source_post_url": "",
  "is_sticker": 0,
  "import_datetime": "2017-12-19 20:43:33",
  "trending_datetime": "0000-00-00 00:00:00",
  "user": {
    "avatar_url": "https://media4.giphy.com/avatars/ladypat/r9JGdMZAeZ6k.gif",
    "banner_url": "https://media4.giphy.com/headers/ladypat/A4x4RZZgNNBE.gif",
    "banner_image": "https://media4.giphy.com/headers/ladypat/A4x4RZZgNNBE.gif",
    "profile_url": "https://giphy.com/ladypat/",
    "username": "ladypat",
    "display_name": "ladypat"
  },
  "images": {
    "fixed_height_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "492",
      "height": "200",
      "size": "39606"
    },
    "original_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "39496"
    },
    "fixed_width": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "200",
      "height": "81",
      "size": "16617",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "9217",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "15664"
    },
    "fixed_height_small_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "246",
      "height": "100",
      "size": "11509"
    },
    "fixed_height_downsampled": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200_d.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "492",
      "height": "200",
      "size": "97481",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200_d.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "48558"
    },
    "preview": {
      "width": "480",
      "height": "194",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-preview.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "30289"
    },
    "fixed_height_small": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "246",
      "height": "100",
      "size": "25650",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "12929",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "20806"
    },
    "downsized_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-downsized_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "39496"
    },
    "downsized": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-downsized.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "101793"
    },
    "downsized_large": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "101793"
    },
    "fixed_width_small_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100w_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "100",
      "height": "41",
      "size": "3105"
    },
    "preview_webp": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-preview.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "460",
      "height": "187",
      "size": "49752"
    },
    "fixed_width_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "200",
      "height": "81",
      "size": "7995"
    },
    "fixed_width_small": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100w.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "100",
      "height": "41",
      "size": "5485",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100w.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "4448",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100w.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "4770"
    },
    "downsized_small": {
      "width": "480",
      "height": "194",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-downsized-small.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "30289"
    },
    "fixed_width_downsampled": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w_d.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "200",
      "height": "81",
      "size": "16617",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200w_d.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "15664"
    },
    "downsized_medium": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "101793"
    },
    "original": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195",
      "size": "101793",
      "frames": "3",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "29006",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "51782",
      "hash": "0d801950c5aa85bdbd8146764a31f9b8"
    },
    "fixed_height": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "492",
      "height": "200",
      "size": "97481",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "28932",
      "webp":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/200.webp?cid=e1bb72ff5b6c7ba658534741452ec414",
      "webp_size": "48558"
    },
    "hd": {
      "width": "800",
      "height": "324",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-hd.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "71418"
    },
    "looping": {
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-loop.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "1493396"
    },
    "original_mp4": {
      "width": "480",
      "height": "194",
      "mp4":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy.mp4?cid=e1bb72ff5b6c7ba658534741452ec414",
      "mp4_size": "29006"
    },
    "preview_gif": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/giphy-preview.gif?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "352",
      "height": "143",
      "size": "48909"
    },
    "480w_still": {
      "url":
        "https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/480w_s.jpg?cid=e1bb72ff5b6c7ba658534741452ec414",
      "width": "480",
      "height": "195"
    }
  },
  "title": "mr blobby 90s GIF by ladypat",
  "_score": 1311117.4
}
