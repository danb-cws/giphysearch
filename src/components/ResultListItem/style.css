.ResultList--item {
  list-style: none;
  width: 16.666667%;
  padding: 0 var(--secondary-horizontal-gutter) 0 0;
  box-sizing: border-box;
}
@media screen and (orientation: portrait) {
  .ResultList--item {
    width: 33.333333%;
    max-height: 50%;
  }
  .ResultList--item:nth-child(-n+3) {
    padding-bottom: var(--secondary-vertical-gutter);
  }
}
.ResultList--item--link {
  display: block;
  box-sizing: border-box;
  height: 100%;
  padding: var(--secondary-vertical-gutter) var(--secondary-horizontal-gutter);
  color: var(--secondary-color);
  background-color: var(--secondary-bg-color);
  cursor: pointer;
}
.ResultList--item--img {
  max-width: 100%;
  max-height: 100%;
  object-fit: contain;
  background: var(--primary-bg-color) var(--spinner-b64-url) no-repeat center center;
  background-size: 1rem;
}
.ResultList--item--title {
  overflow-x: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-size: .83rem;
  line-height: 1.2;
  margin: 0.2rem 0;
}
.ResultList--item--link__selected-item {
  color: var(--highlight-color);
  background-color: var(--highlight-bg-color);
  cursor: default;
}