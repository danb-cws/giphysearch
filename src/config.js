export const GIPHY_API_KEY = 'dc6zaTOxFJmzC';
export const GIPHY_ENDPOINT = 'https://api.giphy.com/v1/gifs/search?q=';
export const MAX_RESULTS = 5;
export const default_term = 'grumpy cat';