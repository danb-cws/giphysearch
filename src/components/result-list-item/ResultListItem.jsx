import React from "react";
import "./ResultListItem.css";

const ResultListItem = ({ gif, selectedItem, index, onItemSelect }) => {
  return (
    <li className="ResultList--item">
      <a
        className={selectedItem === index ? "ResultList--link__selected-item" : "ResultList--link"}
        onClick={() => {
          onItemSelect(index);
        }}
      >
        <img
          className='ResultList--item--img'
          src={gif.images.fixed_height_small_still.url}
          width={gif.images.fixed_height_small_still.width}
          height={gif.images.fixed_height_small_still.height}
          alt={gif.title}
        />
        <h5 className="ResultList--item--title">{gif.title}</h5>
      </a>
    </li>
  );
};

export default ResultListItem;
