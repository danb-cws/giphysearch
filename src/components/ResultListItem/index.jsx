import React from "react";
import "./style.css";

const ResultListItem = ({ gif, selectedItem, index, onItemSelect, ChildNodeName }) => {
  return (
    <ChildNodeName className="ResultList--item">
      <a
        className={selectedItem === index ? "ResultList--item--link ResultList--item--link__selected-item" : "ResultList--item--link"}
        onClick={() => {
          onItemSelect(index);
        }}
      >
        <img
          className='ResultList--item--img'
          src={gif.images.fixed_height_small_still.url}
          width={gif.images.fixed_height_small_still.width}
          height={gif.images.fixed_height_small_still.height}
          alt={gif.title}
        />
        <h5 className="ResultList--item--title">{gif.title || '(No title)'}</h5>
      </a>
    </ChildNodeName>
  );
};

export default ResultListItem;
