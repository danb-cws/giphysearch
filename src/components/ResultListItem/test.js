import React from "react";
import { shallow } from "enzyme";
import ResultListItem from "./";
import testjson from "mock_json/gifMock.json";
import * as config from "config";

const elSelected = shallow(
  <ResultListItem
    gif={testjson}
    selectedItemIndex={config.RESULTS_PER_PAGE + 2}
    index={2}
    resultsPageIndex={1}
    onItemSelect={() => {
      return true;
    }}
    ChildNodeName={"li"}
  />
);

/*const elUnselected = shallow(
  <ResultListItem
    gif={testjson}
    selectedItemIndex={config.RESULTS_PER_PAGE + 2}
    index={3}
    resultsPageIndex={1}
    onItemSelect={() => {
      return true;
    }}
    ChildNodeName={"li"}
  />
);*/

describe("ResultListItem tests", () => {
  it("Renders the component", () => {
    expect(elSelected.exists()).toBe(true);
  });

  it("Correctly adds class to the selected item", () => {
    expect(
      elSelected
        .find(".ResultList--item--link")
        .hasClass("ResultList--item--link__selected-item")
    ).toBe(true);
  });

/* TODO revisit this */

/*  it("Actions click on item", () => {
    expect(
      elUnselected.find(".ResultList--item--link").simulate("click", { preventDefault(){} })
    ).toBeTruthy();
  });*/

});
