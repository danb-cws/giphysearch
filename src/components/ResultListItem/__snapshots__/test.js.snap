// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultListItem tests Matches snapshot 1`] = `
<li
  className="ResultList--item"
>
  <a
    className="ResultList--item--link ResultList--item--link__selected-item"
    href="https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414"
    onClick={[Function]}
  >
    <img
      alt="mr blobby 90s GIF by ladypat"
      className="ResultList--item--img"
      src="https://media1.giphy.com/media/3ohc1f1vrhZsufbBcI/100_s.gif?cid=e1bb72ff5b6c7ba658534741452ec414"
    />
    <h5
      className="ResultList--item--title"
    >
      mr blobby 90s GIF by ladypat
    </h5>
  </a>
</li>
`;
